air/src/air/transition/frame.rs

```diff
@@ -3,80 +3,96 @@
 // This source code is licensed under the MIT license found in the
 // LICENSE file in the root directory of this source tree.
 
-use super::{FieldElement, Vec};
+use super::{super::Air, FieldElement, Vec};
+use crate::Table;
+use utils::TableReader;
 
-// EVALUATION FRAME
-// ================================================================================================
 /// A set of execution trace rows required for evaluation of transition constraints.
-///
-/// In the current implementation, an evaluation frame always contains two consecutive rows of the
-/// execution trace. It is passed in as one of the parameters into
+/// It is passed in as one of the parameters into
 /// [Air::evaluate_transition()](crate::Air::evaluate_transition) function.
+pub trait EvaluationFrame<E: FieldElement> {
+    const FRAME_OFFSETS: [usize; 2] = [0, 1];
+    const FRAME_SHIFT: usize = 1;
+
+    /// Creates an empty frame
+    fn new<A: Air>(air: &A) -> Self;
+
+    /// Creates a new frame instantiated from the provided row-major table
+    fn from_table(table: Table<E>) -> Self;
+
+    /// Convert frame to a row-major table
+    fn to_table(&self) -> Table<E>;
+
+    /// Reads selected trace rows from the supplied data into the frame
+    fn read_from<R: TableReader<E>>(&mut self, data: R, step: usize, blowup: usize);
```

Comment:
What is `blowup` here? What would be the blowup in the original case of two consecutive rows?